lib/ansible/modules/net_tools/basics/uri.py

```diff
@@ -317,6 +323,23 @@ def absolute_location(url, location):
         return location
 
 
+def form_urlencoded(body):
+    ''' Convert data into a form-urlencoded string '''
+    if isinstance(body, string_types):
+        return body
+    elif hasattr(body, '__iter__'):
```

Comment:
Why are we doing this `hasattr` here?  Generally speaking, we should use `hasattr` as little as possible.

Reply:
Sure, testing for a dict seems more appropriate. I also wonder whether we ought to support lists in some way.

Reply:
I'd recommend having an opinion that it be a dict.  If it is a list of some sort, you can use the jinja2 dict constructor first.

Reply:
So this is now fixed.

Reply:
Both sequence and mappings are supported, just like the implementation from requests which supports this as well.

Reply:
Let me add some tests for both list and dict input, and test invalid input too.